modules/slowzones/map/SlowSegmentLabel.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useMemo } from 'react';
+
+import type { SlowZoneResponse } from '../../../common/types/dataPoints';
+import type { LineMetadata } from '../../../common/types/lines';
+import { stringifyTime } from '../../../common/utils/time';
+
+import type { SlowZoneDirection, SlowZonesByDirection } from './segment';
+import { DIRECTIONS } from './segment';
+
+import styles from './SlowSegmentLabel.module.css';
+import DirectionIndicator from './DirectionIndicator';
+
+type SlowSegmentLabelProps = {
+  slowZonesByDirection: SlowZonesByDirection;
+  line: LineMetadata;
+  isHorizontal: boolean;
+};
+
+type SlowZoneLabelProps = {
+  direction: SlowZoneDirection;
+  slowZone: SlowZoneResponse;
+  color: string;
+  isHorizontal: boolean;
+};
+
+const SlowZoneLabel = (props: SlowZoneLabelProps) => {
+  const {
+    direction,
+    isHorizontal,
+    color,
+    slowZone: { delay },
+  } = props;
+
+  const delayString = useMemo(
+    () =>
+      stringifyTime(delay, {
+        showHours: false,
+        showSeconds: true,
+        truncateLeadingMinutesZeros: true,
+      }),
+    [delay]
+  );
+
+  return (
+    <div
+      style={{ flexDirection: isHorizontal && direction === '0' ? 'row-reverse' : 'row' }}
+      className={styles.slowZoneLabel}
+    >
+      <DirectionIndicator
+        direction={direction}
+        color={color}
+        isHorizontal={isHorizontal}
+        size={2}
+      />
+      {delayString}
+    </div>
+  );
+};
+
+const SlowSegmentLabel = (props: SlowSegmentLabelProps) => {
```

Comment:
In modern react, if you should use the `React.FC` or `React.FunctionComponent` type, with the props interface passed into that typing.
```suggestion
const SlowSegmentLabel: React.FC<SlowSegmentLabelProps> = (props) => {
```

We also should use object spreading for the props and avoid `props.` when possible.

```
const SlowSegmentLabel: React.FC<SlowSegmentLabelProps> = ({ slowZonesByDirection, line, isHorizontal }) => {
```